lara-typescript/src/section-authoring/components/section-item-picker.tsx

```diff
@@ -1,45 +1,158 @@
-import * as React from "react";
+import React, { useState } from 'react';
+
+import "./section-item-picker.css"
 
 export interface ItemType {
-  name: string
+  name: string,
+  use_count: number,
+  date_added: number
 }
 
 export interface IProps {
   availableItemTypes: ItemType[],
-  quickAddItems: ItemType[]
+  quickAddItems: ItemType[],
+  allItems: ItemType[],
+  matchingItems: ItemType[]
 }
 
 export const SectionItemPicker: React.FC<IProps> = (props) => {
-  const { quickAddItems } = props;
+  const { allItems, quickAddItems } = props;
+  let { matchingItems } = props;
+  const [itemSelected, setItemSelected] = useState(false);
+  const [currentSelectedItem, setCurrentSelectedItem] = useState("");
+  const [allItemsList, setAllItemsList] = useState(allItems);
+  const [isSearching, setIsSearching] = useState(false);
+
+  const setItemClasses = (isSelectedItem: boolean) => {
+    const classes = isSelectedItem ? "assessmentItemOption selected" 
+                                   : currentSelectedItem !== "" 
+                                     ? "assessmentItemOption disabled"
+                                     : "assessmentItemOption";
```

Comment:
This is the kind of code that warrants the use of `classNames()`:
```suggestion
    const classes = classNames("assessmentItemOption", { selected: isSelectedItem, disabled: currentSelectedItem !== "" });
```
It looks like `classNames()` is not already being included, but I'd be inclined to add it since it's small and likely to be useful elsewhere in the `lara-typescript` project.

Reply:
@ksweson Thanks 👍  Seems like a good idea.